packages/core/utils/src/validate/visitors/throw-unrecognized-fields.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  isDynamicZoneAttribute,
+  isMorphToRelationalAttribute,
+  isRelationalAttribute,
+  constants,
+  isComponentSchema,
+  isMediaAttribute,
+} from '../../content-types';
+import type { Visitor } from '../../traverse-entity';
+import { throwInvalidKey } from '../utils';
+
+// TODO these should all be centralized somewhere instead of maintaining a list
+const ID_FIELDS = [constants.DOC_ID_ATTRIBUTE, constants.DOC_ID_ATTRIBUTE];
+const ALLOWED_ROOT_LEVEL_FIELDS = [...ID_FIELDS];
+const MORPH_TO_ALLOWED_FIELDS = ['__type'];
+const DYNAMIC_ZONE_ALLOWED_FIELDS = ['__component'];
+const RELATION_REORDERING_FIELDS = ['connect', 'disconnect', 'set', 'options'];
+
+const throwUnrecognizedFields: Visitor = ({ key, attribute, path, schema, parent }) => {
+  // We only look at properties that are not attributes
+  if (attribute) {
+    return;
+  }
+
+  // At root level (path.attribute === null), only accept allowed fields
+  if (path.attribute === null) {
+    if (ALLOWED_ROOT_LEVEL_FIELDS.includes(key)) {
+      return;
+    }
+
+    return throwInvalidKey({ key, path: attribute });
+  }
+
+  // allow special morphTo keys
+  if (isMorphToRelationalAttribute(parent?.attribute) && MORPH_TO_ALLOWED_FIELDS.includes(key)) {
+    return;
+  }
+
+  // allow special dz keys
+  if (
+    isComponentSchema(schema) &&
+    isDynamicZoneAttribute(parent?.attribute) &&
+    DYNAMIC_ZONE_ALLOWED_FIELDS.includes(key)
+  ) {
+    return;
+  }
+
+  // allow special relation reordering keys
+  // TODO: Only toMany or all?
```

Comment:
@Marc-Roig is relation reordering enabled for toMany relations only? Or should it be possible to have it for any kind relation?

Reply:
Only manytoX and XToMany relations have ordering